File: linear_search/c/linear_search.c
#include <stdio.h>

// Function to perform linear search
int linearSearch(int arr[], int n, int target) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) {
            return i; // Return the index of the target element
        }
    }
    return -1; // Return -1 if the target is not found
}

int main() {
    int arr[] = {34, 78, 12, 54, 89, 23}; // Array to search in
    int n = sizeof(arr) / sizeof(arr[0]); // Calculate the number of elements in the array
    int target = 54; // Target value to search for

    int result = linearSearch(arr, n, target);
    if (result != -1) {
        printf("Element %d found at index %d.\n", target, result);
    } else {
        printf("Element %d not found in the array.\n", target);
    }

    return 0;
}

File: linear_search/python/linear_search.py
def linear_search(arr, target):
    """
    This function performs a linear search on an array.
    
    Parameters:
    arr (list): The list of elements to search through.
    target (any): The target value to search for.

    Returns:
    int: The index of the target if found, otherwise -1.
    """
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

# Example usage:
elements = [5, 3, 7, 1, 9, 8]
target_value = 7

# Perform the search
index = linear_search(elements, target_value)

# Output the result
if index != -1:
    print(f"Target found at index {index}")
else:
    print("Target not found")
